File: examples/algorithms/cross-language-hook/choose.py
from gear_algorithm import component

@component(id="choose", input_schema={"type":"object","properties":{"options":{"type":"array","items":{"type":"string"}}},"required":["options"],"additionalProperties":False}, output_schema={"type":"object","properties":{"choice":{"type":"string"}},"required":["choice"],"additionalProperties":False}, scope="campaign")
def choose(value):
    return {"choice": value["options"][0]}

File: examples/algorithms/ts-toy/choose.mjs
export const choose = {
  describe() {
    return {
      id: 'choose',
      scope: 'campaign',
      inputSchema: { type: 'object', properties: { options: { type: 'array', items: { type: 'string' } } }, required: ['options'], additionalProperties: false },
      outputSchema: { type: 'object', properties: { choice: { type: 'string' } }, required: ['choice'], additionalProperties: false },
    };
  },
  invoke(value) {
    return { choice: value.options[0] };
  },
};
